Updating documentation for REST switch with template example

cribbstechnologies opened the pull request:
**Description:**

I've updated the documentation to include an example with `is_on_template` and fixed a typo in the current docs

Landrash commented on source/_components/switch.rest.markdown:
```diff
@@ -30,10 +30,34 @@ Configuration variables:
+switch:
+  - platform: rest
+    resource: http://<address>/led_endpoint
+    body_on: '{"active": "true"}
```
Is the ' there on purpose or are we lacking one?

Landrash commented on source/_components/switch.rest.markdown:
```diff
@@ -30,10 +30,34 @@ Configuration variables:
+  - platform: rest
+    resource: http://<address>/led_endpoint
+    body_on: '{"active": "true"}
+    body_off: '{"active": "false"}
```
Is the ' there on purpose or are we lacking one?

fabaff commented on source/_components/switch.rest.markdown:
```diff
@@ -30,10 +30,34 @@ Configuration variables:
+<p class='note info'>
+When Home Assistant is trying to update the state of the device by issuing a GET to the endpoint, the value returned from the endpoint is made available to the template as `value_json`. Values within the JSON can be referenced using dot notation:
+`{"state": "off"}` would be referenced as `{% raw %}{{value_json.state}}{% endraw $}`
+</p>
```
The info part looks like the re-phrased details from `is_on_template` and we should start to add details to the platforms which are already mentioned in the template documentation. If it's unclean them the template documentation should be expanded.

cribbstechnologies replied:
I specifically asked @balloob if I should add this info in some overarching place and he said if it wasn't in the component people wouldn't find it. I've pushed this block into the is_on_template definition but it looks crowded now.

fabaff replied:
I don't know how many platforms are supporting templates nowadays almost all contains a link to the template documentation. The details for accessing/processing incoming data is a topic that concerns very platform and is not limited to REST.

fabaff commented on source/_components/switch.rest.markdown:
```diff
@@ -30,10 +30,34 @@ Configuration variables:
+
+### {% linkable_title Switch with templated value %}
+
+This example shows a switch that uses a [template](/topics/templating/) to allow Home Assistant to determine its state. In this example the REST endpoint returns this JSON response with true indicating the switch is on
```
Period is missing at the end.

cribbstechnologies commented:
I didn't mean to do this and don't know what caused it.

On Fri, May 26, 2017 at 2:32 PM, Fabian Affolter <notifications@github.com>
wrote:

> *@fabaff* requested changes on this pull request.
>
> Please don't change the file permissions (644->755).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant.github.io/pull/2713#pullrequestreview-40590806>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABjyfRD_XscjhcGneiIKbrvGmiurS9CFks5r9xqggaJpZM4NmhmB>
> .
>

cribbstechnologies commented:
The link to the template page is already there. I removed the extra
explanation but I've now answered three questions in the community about
REST-related components and how to parse the values in the template. It's
not readily obvious to newcomers that value_json is how the data will be
returned and not just an example. I'm unsure how this should best be
addressed.

On Fri, May 26, 2017 at 2:30 PM, Fabian Affolter <notifications@github.com>
wrote:

> *@fabaff* commented on this pull request.
> ------------------------------
>
> In source/_components/switch.rest.markdown
> <https://github.com/home-assistant/home-assistant.github.io/pull/2713#discussion_r118764072>
> :
>
> >
> +<p class='note info'>
> +When Home Assistant is trying to update the state of the device by issuing a GET to the endpoint, the value returned from the endpoint is made available to the template as `value_json`. Values within the JSON can be referenced using dot notation:
> +`{"state": "off"}` would be referenced as `{% raw %}{{value_json.state}}{% endraw $}`
> +</p>
>
> I don't know how many platforms are supporting templates nowadays almost
> all contains a link to the template documentation. The details for
> accessing/processing incoming data is a topic that concerns very platform
> and is not limited to REST.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant.github.io/pull/2713#discussion_r118764072>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABjyfYkbIFJIN2uTrcDTlPdJjaNOfLnJks5r9xpSgaJpZM4NmhmB>
> .
>
